### PR TITLE
Dt 7256 ignored razor expressions environment variable only works from command line

### PR DESCRIPTION
**Pull request by jocgir**
This pull request includes various changes to improve documentation, correct typos, and enhance functionality in several files. The most important changes include adding a new ignore words list to the `.cspell.json` file, fixing typos across multiple files, and enhancing the `applyRazor` function for better regex handling.

### Documentation and Comments:

* [`.cspell.json`](diffhunk://#diff-4e0f175c840784de9e935f881821905c75a2a125aa57c51268a7fb4496edfbe6R1-R157): Added a comprehensive list of words to ignore for spell checking.
* [`template/doc.go`](diffhunk://#diff-9cb8d5e371253e4cc89e4667d5e60c1347144772fe0d9660d2488e120703a408R1-R12): Added a detailed package-level comment to describe the functionalities provided by the `template` package.

### Typo Fixes:

* [`collections/implementation/generic_test.go`](diffhunk://#diff-d4fb5da892d99befdf262f60a50c7793b7057f7e6e780778b2a35847a8d1325aL700-R700): Corrected "Non existant" to "Non existent" in multiple test cases. [[1]](diffhunk://#diff-d4fb5da892d99befdf262f60a50c7793b7057f7e6e780778b2a35847a8d1325aL700-R700) [[2]](diffhunk://#diff-d4fb5da892d99befdf262f60a50c7793b7057f7e6e780778b2a35847a8d1325aL725-R727) [[3]](diffhunk://#diff-d4fb5da892d99befdf262f60a50c7793b7057f7e6e780778b2a35847a8d1325aL753-R755)
* [`hcl/generated_test.go`](diffhunk://#diff-07a22d936109fa7ab850eff11982f321040858458f9008d4f88718698819d5aeL704-R704): Corrected "Non existant" to "Non existent" in multiple test cases. [[1]](diffhunk://#diff-07a22d936109fa7ab850eff11982f321040858458f9008d4f88718698819d5aeL704-R704) [[2]](diffhunk://#diff-07a22d936109fa7ab850eff11982f321040858458f9008d4f88718698819d5aeL729-R731) [[3]](diffhunk://#diff-07a22d936109fa7ab850eff11982f321040858458f9008d4f88718698819d5aeL757-R759)
* [`json/generated_test.go`](diffhunk://#diff-8c8abc56c96d53f23e494a600059a965d6d129186102936ed8fee3747267dc72L704-R704): Corrected "Non existant" to "Non existent" in multiple test cases. [[1]](diffhunk://#diff-8c8abc56c96d53f23e494a600059a965d6d129186102936ed8fee3747267dc72L704-R704) [[2]](diffhunk://#diff-8c8abc56c96d53f23e494a600059a965d6d129186102936ed8fee3747267dc72L729-R731) [[3]](diffhunk://#diff-8c8abc56c96d53f23e494a600059a965d6d129186102936ed8fee3747267dc72L757-R759)
* [`template/extra_math.go`](diffhunk://#diff-c388a8fc58ef0010edf280fa3da269f37fb9840ddef6a0f9c1fc74d2167af578L294-R294): Corrected "substraction" to "subtraction".
* [`template/extra_runtime.go`](diffhunk://#diff-da65b666c2f2215524381a459dbca8e5c9ebb70b0e41cb3cec3c3d3b39a545c9L55-R56): Corrected multiple instances of "formated" to "formatted". [[1]](diffhunk://#diff-da65b666c2f2215524381a459dbca8e5c9ebb70b0e41cb3cec3c3d3b39a545c9L55-R56) [[2]](diffhunk://#diff-da65b666c2f2215524381a459dbca8e5c9ebb70b0e41cb3cec3c3d3b39a545c9L97-R97) [[3]](diffhunk://#diff-da65b666c2f2215524381a459dbca8e5c9ebb70b0e41cb3cec3c3d3b39a545c9L368-R369)
* [`template/extra_utils.go`](diffhunk://#diff-9d727b480758b3178ce7e242aeb15d4d7bc6d43845a3c82cc44f613a2395fa34L87-R87): Corrected "forgeround" to "foreground".
* [`template/func_table.go`](diffhunk://#diff-9b3955484b1d7122995b17aaca76181b54c4ce011ddbc97ce8608cce8a5a7cbbL205-R205): Corrected "detailled" to "detailed".

### Functionality Enhancements:

* [`template/razor.go`](diffhunk://#diff-0fd1c6f53450fc9d60b7123dc28c3986c89f32fb72d1e7afb089714d3450a4f2L20-R24): Improved the `applyRazor` function to ensure regex is not greedy and stops at the end of the expression.
* [`template/razor_test.go`](diffhunk://#diff-8de2928fadf691929b426fcaa1b802aefc24091fa551948e9bd3ac37ac6ee2daR150-R223): Added new test cases for `applyRazor` function to handle strict error checking and various scenarios.

### Miscellaneous:

* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R72-R73): Added a link to the library usage documentation.
* [`main.go`](diffhunk://#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261L37-R37): Updated the documentation URL.
* [`render-doc`](diffhunk://#diff-55ed54c0cc131de434de6bb3c903384ce34417117e16fd429dd83f0e55127924L132-R132): Modified the script to exclude specific lines from the README.

### CODE REVIEW

File: .cspell.json [hunk 1]
```diff
@@ -0,0 +1,157 @@
+{
+    "ignoreWords": [
+        "abaabaaa",
+        "abaabaaabaaaa",
```
Comment (jonapich) [inline .cspell.json:4]: image

File: template/razor.go [hunk 2]
```diff
@@ -17,7 +17,11 @@ func (t *Template) applyRazor(content []byte) (result []byte, changed bool) {
 	t.ensureInit()
 
 	for _, ignoredExpr := range t.ignoredRazorExpr {
-		content = regexp.MustCompile(t.delimiters[2]+ignoredExpr).ReplaceAllFunc(content, func(match []byte) []byte {
+		if strings.HasSuffix(ignoredExpr, "*") || strings.HasSuffix(ignoredExpr, "+") {
+			ignoredExpr += `?` // Ensure that the regex is not greedy
+		}
+		ignoredExpr = t.delimiters[2] + ignoredExpr + `\b` // Stop at the end of the expression
```
Comment (jonapich) [inline template/razor.go:23]: je pense que le `t.delimiters[2]` mérite un petit commentaire; j'ai du fouiller un peu pour catcher que c'est like `['{{', '}}', '@']`.

File: template/template.go [hunk 3]
```diff
@@ -381,3 +338,19 @@ func (t *Template) Context() (result collections.IDictionary) {
...
+//	expr: A variadic parameter representing one or more Razor expressions to be ignored.
+func (t *Template) AppendIgnoreRazorExpression(expr ...string) {
+	t.ignoredRazorExpr = append(t.ignoredRazorExpr, expr...)
+}
+
+// GetIgnoredRazorExpressions returns a slice of strings containing the ignored Razor expressions.
+// These expressions are not processed by the template engine.
+func (t Template) GetIgnoredRazorExpressions() []string {
+	return t.ignoredRazorExpr
+}
```
Comment (jonapich) [inline template/template.go:356]: pas utilisé? 🤔

**Comment by jonapich** [comment]
Mon idea me montre des warnings un peu partout: 
image

File: template/template.go (hunk 3 above)
Comment (jocgir) [reply template/template.go]: C'est une librairie, ça n'est pas forcément utilisé dans la librarie.

File: .cspell.json (hunk 1 above)
Comment (jocgir) [reply .cspell.json:4]: J'ai un spell checker et comme c'est une string qui est dans le code, il me la rapporte comme une erreur

File: template/razor.go (hunk 2 above)
Comment (jocgir) [reply template/razor.go:27]: Yes, you are right. Good catch

File: template/template.go (hunk 3 above)
Comment (jocgir) [reply template/template.go]: J'ai ajouté des tests qui les utilisent

**Comment by jocgir** [comment]
C'est discutable, mais je l'ai changé.

**Review by jonapich — APPROVED** [review]

**Pull request merged**